common/src/main/java/fr/rakambda/fallingtree/common/tree/IBreakAttemptResult.java

```diff
@@ -0,0 +1,14 @@
+package fr.rakambda.fallingtree.common.tree;
+
+import fr.rakambda.fallingtree.common.wrapper.IBlockPos;
+import fr.rakambda.fallingtree.common.wrapper.ILevel;
+import fr.rakambda.fallingtree.common.wrapper.IPlayer;
+
+/**
+ * The result of a {@link TreeHandler#attemptTreeBreaking(ILevel, IPlayer, IBlockPos)}, whether it succeeded or not.
+ * Failures are generally instances of {@link BreakAbortionCause}, where are succeeded attempts are instances of
+ * {@link BreakTreeResult}.
+ */
+public interface IBreakAttemptResult{
```

Comment:
I think we can turn this into a `sealed` interface, this way in `BlockBreakListener` we can also use return-switch that'll fail compilation if another type is later added

Reply:
I am not exactly sure what you mean with return-switch, probably given that I haven't spend a lot of time with the new switch patterns as I primarily work with Java 8.

Reply:
This is apparently not possible in Java 17, my bad. I refactored the shouldCancel method instead, to bring the logic inside the class itself.